File: Data Structures & Algorithms/counting-bits/submission-1.cpp
class Solution {
    public:
        vector<int> countBits(int n) {
                vector<int> out(n+1);
                for(int i = 0; i <= n; i++){
                        out[i] = out[i >>1 ] + (i & 1);
                }
                return out;
        };
};

File: Data Structures & Algorithms/find-duplicate-integer/submission-1.rs
impl Solution {
    pub fn find_duplicate(mut array: Vec<i32>) -> i32 {
  let mut count =0;
      while count<array.len(){
              let fe = array[0];
                      if fe == array[fe as usize]{
                                  return fe;
                                          }else{
                                                      array.swap(0,fe as usize);
                                                              }
                                                                      count+=1;
                                                                          }
                                                                              0
    }
}
